qcfractal/interface/models/gridoptimization.py

```diff
@@ -263,28 +264,31 @@ def deserialize_key(self, key:str)->Tuple[int]:
         """
         return tuple(json.loads(key))
 
-    def get_scan_value(self, scan_number: int) -> Tuple[List[float]]:
+    def get_scan_value(self, scan_number: Union[str, int, Tuple[int]]) -> Tuple[List[float]]:
         """
         Obtains the scan parameters at a given grid point.
 
         Parameters
         ----------
-        key : Union[str, int, Tuple[int]]
+        scan_number : Union[str, int, Tuple[int]]
             The key of the scan.
 
         Returns
         -------
         Tuple[List[float]]
             Description
         """
-        if isinstance(key, str):
-            key = self.deserialize_key(key)
+        if isinstance(scan_number, str):
+            scan_number = self.deserialize_key(scan_number)
 
         ret = []
-        for n, idx in enumerate(key):
+        for n, idx in enumerate(scan_number):
             ret.append(self.keywords.scans[n].steps[idx])
 
-        return tuple(ret)
+        # Cast to Tuple[List[float]] instead of tuple(ret) -> Tuple[float,...]
+        ret = (ret, )
```

Comment:
This function would have failed due to undefined values, but I made the return match the call.